internal/app/machined/main.go

```diff
@@ -157,31 +158,48 @@ func main() {
 			rebootFlag = unix.LINUX_REBOOT_CMD_POWER_OFF
 			fallthrough
 		case event.Reboot:
-			if err := seq.Shutdown(); err != nil {
-				panic(fmt.Errorf("shutdown failed: %w", err))
-			}
+			go func() {
+				if err := seq.Shutdown(); err != nil {
+					if errors.Is(err, sequencerv1alpha1.ErrLocked{}) {
+						log.Printf("failed to run shutdown sequence: %v", err)
 
-			sync()
+						return
+					}
 
-			if unix.Reboot(rebootFlag) == nil {
-				select {}
-			}
+					panic(err)
+				}
+
+				sync()
+
+				if unix.Reboot(rebootFlag) == nil {
+					select {}
+				}
+			}()
 		case event.Upgrade:
-			var (
-				req *machineapi.UpgradeRequest
-				ok  bool
-			)
+			go func() {
```

Comment:
Prevents upgrade requests from queuing up.

Reply:
on other hand, if we pass all the events which launch sequences via the event bus, doesn't it mean we effectively achieve one-at-a-time lock?

we could do something like

```go
func (bus) TryPublish(timeout, event) error {
   select {
   case bus <- event:
      return nil
   case <-time.After(timeout):
     return fmt.Errorf("bus is locked by another activity, please try later)
   }
}
```